xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -117,29 +117,27 @@ public void createXdsClientAndStart() throws IOException {
       throw new IOException("missing grpc_server_resource_name_id value in xds bootstrap");
     }
     XdsClient xdsClientImpl =
-        new ServerXdsClient(
+        new ClientXdsClient(
```

Comment:
I guess using a shared XdsClient instance for multiple gRPC servers in the same process is also one of the goals. Then you should use the global `SharedXdsClientPoolProvider` to obtain the XdsClient instance instead of creating one every time.

Anyway, that might be separate than this PR. I believe you would probably need more restructuring (e.g., accessing bootstrap config) before switching to that.

Reply:
Yes, as discussed offline and described in my multi-step approach to use ClientXdsClient on the server side, using `SharedXdsClientPoolProvider` comes later and is not going to be part of this PR. I thought I had shared the doc but I will do it.